default/python/AI/fleet_orders.py

```diff
@@ -115,15 +118,38 @@ def can_issue_order(self, verbose=False):
         threat = f_threat + m_threat + p_threat
         safety_factor = foAI.foAIstate.character.military_safety_factor()
         universe = fo.getUniverse()
+        if main_fleet_mission.type == MissionType.INVASION:
+            # Don't advance outside of our fleet-supply zone unless the target either has no shields at all or there
+            # is already a military fleet assigned to secure the target, and don't take final jump unless the planet is
+            # (to the AI's knowledge) down to zero shields.  Additional checks will also be done by the later
+            # generic movement code
+            invasion_planet = main_fleet_mission.target
+            inv_plt_obj = invasion_planet.get_object()
```

Comment:
Please do not abbreviate variable names like this.

We generally refer to the "planet_object" as "planet" in all other parts of the code. How about:
```invasion_planet``` --> ```invasion_target```
```inv_plt_obj``` -->  ```invasion_planet```
```invasion_sys``` -->```invasion_system```